### PR TITLE
Update thread fields in utils.formatThread

### LINKED ISSUE #485: getThreadInfo formerParticipants is null
After removing a user from a group, formerParticipants for that group is still null. How is this property updated and does Facebook even give this information? I tried looking into this myself but I wouldn't know how to test it and got confused since getThreadInfo uses getUserInfo but getUserInfo's info does have the same properties.

Thanks!

### PR DESCRIPTION
**Pull request by mangopearapples**
This PR changes some of the properties in `formatThread`. I removed some deprecated fields and added the missing fields that were available in the payload but not to the user.

This also somewhat resolves #485 by removing the field from the DOCS and `formatThread`

Notable changes:

- `muteSettings` -> `muteUntil`: is the timestamp for which the thread is muted until. -1 is forever.
- -`canonicalFbid`
- +`lastMessageTimestamp` + `lastReadTimestamp`
- +`adminIDs`
- +`threadType`

### CODE REVIEW

**Comment by Schmavery** [comment]
Thanks for updating this, @mangopearapples 
As always when we're updating the public interface like this, is it possible to get any more info about what is in these new fields?  It would be amazing if we could update the docs to have this information too, but that might be out of scope for this change (I would give lots of emoji hearts to whoever takes on that task).
If facebook has stopped serving some of this data, I'm ok to remove it, but we should try to be at least a little careful not to add more data people won't know how to use?

**Comment by mangopearapples** [comment]
You're right about adding data people won't know how to use. Should I remove them? I had somewhat considered this and ignored some properties like `ephemeral_ttl_mode` but I though fields like adminIDs and `lastMessageTimestamp` were self explanatory. 

Are there any particular fields you should not be added?

I will update docs to include description.

**Comment by Schmavery** [comment]
isPinProtected/adminIDs/threadType/reactionsMuteMode/mentionsMuteMode
Maybe these correspond to thread features I'm unaware of?

**Comment by Schmavery** [comment]
Also, searching through the repo, it looks like formatThread is used by 3 functions: getThreadList, getThreadInfo, and searchForThread.  Which of these were you able to test with the changes?  Do we receive all the values in all these cases?

**Comment by mangopearapples** [comment]
@Schmavery Are doc formats like [this](https://github.com/mangopearapples/facebook-chat-api/blob/hotfix/formatThread-fields-fix/DOCS.md#apigetthreadinfothreadid-callback) okay?

Also from what's available to me, adminIDs are members that are admins in groups in chats belonging to a Facebook group. E.g. .

`threadType` seems to be '1' for normal user chats and '2' for group chats and maybe different values for chats from pages. More testing is needed for this but I think it is useful data for the user.

I was assuming mentions and reactions can be muted in the settings but maybe not. I will remove isPinProtected, reactionsMuteMode, and mentionsMuteMode. Is this okay?

I have tested with `getThreadList` and `getThreadInfo` which give identical values (apart from `getThreadList` being an array of course). I was unaware of `searchForThread` so more testing on that needs to be done.

**Comment by mangopearapples** [comment]
After a bit of research, doesn't look like Facebook uses `search_threads.php` anymore. Instead, it uses GraphQL. Issue #396 seems to correspond with this too. I can't get `searchForThread` to give me any thread data so I can't know if a successful call will give me appropriate values.

**Comment by Schmavery** [comment]
Great, thanks a lot for following up on this.  Threadtype looks very interesting, we were using getUserInfo in a hacky way to get this info, but maybe we can switch over now.  I wonder how much this overlaps with isCanonical (I never knew what a canonical thread was until now).
Thanks for trying out getThreadList and getThreadInfo.  The doc changes are great.

File: DOCS.md [hunk 1]
```diff
@@ -516,9 +516,43 @@ Takes a threadID and a callback.  Works for both single-user and group threads.
...
+| unreadCount | Number of unread messages |
+| messageCount | Number of messages |
+| imageSrc | URL to the group chat photo. Null if unset or a canonical thread. |
+| timestamp |  |
+| serverTimestamp |  |
+| muteUntil | Timestamp at which the thread will no longer be muted. The timestamp will be -1 if the thread is muted indefinitely or null if the thread is not muted. |
+| mentionsMuteMode |  |
+| isCanonicalUser | True if the other user in a canonical thread is a user, false if the other user is a page or group. |
+| isCanonical | True if the thread is a private chat to a single user, false for group chats. |
+| isPinProtected |  |
```
Comment (Schmavery) [inline DOCS.md]: Should this be removed now

Comment (mangopearapples) [inline DOCS.md]: Yep forgot about that one. Done

**Comment by Schmavery** [comment]
Thanks!

**Pull request merged**